handlers/handlers.go

```diff
@@ -0,0 +1,360 @@
+package handlers
+
+import (
+	"bytes"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"log"
+	"net/http"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/Xa4y322/go_final_project/database"
+	"github.com/Xa4y322/go_final_project/models"
+)
+
+const DateFormat string = "20060102"
+
+func nextDate(now time.Time, date string, repeat string) (string, error) {
+	if repeat == "" {
+		return "", errors.New("repeate is empty")
+
+	}
+	if strings.Contains(repeat, "d ") {
+		daysToAdd, err := strconv.Atoi(strings.TrimPrefix(repeat, "d "))
+		if err != nil {
+			return "", err
+		}
+		if daysToAdd > 400 {
+			return "", errors.New("repeat period in days max 400")
+		}
+
+		beginDate, err := time.Parse(DateFormat, date)
+		if err != nil {
+			return "", err
+		}
+
+		newDate := beginDate.AddDate(0, 0, daysToAdd)
+		for newDate.Before(now) {
+			newDate = newDate.AddDate(0, 0, daysToAdd)
+		}
+
+		return newDate.Format(DateFormat), nil
+
+	}
+	if repeat == "y" {
+		beginDate, err := time.Parse(DateFormat, date)
+		if err != nil {
+			return "", err
+		}
+
+		newDate := beginDate.AddDate(1, 0, 0)
+
+		for newDate.Before(now) {
+			newDate = newDate.AddDate(1, 0, 0)
+		}
+
+		return fmt.Sprint(newDate.Format(DateFormat)), nil
+	} else {
+		return "", errors.New("repeat string has wrong format")
+	}
+}
+
+func NextDateGET(w http.ResponseWriter, r *http.Request) {
+	pNow, err := time.Parse(DateFormat, r.FormValue("now"))
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	pDate := r.FormValue("date")
+	pRepeat := r.FormValue("repeat")
+	newDate, err := nextDate(pNow, pDate, pRepeat)
+
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusBadRequest)
+		return
+	}
+
+	w.WriteHeader(http.StatusOK)
+	_, err = w.Write([]byte(newDate))
+
+	if err != nil {
+		http.Error(w, fmt.Sprintf("nextDateGET error: %w", err), http.StatusBadRequest)
+	}
+}
+
+func responseWithError(w http.ResponseWriter, err error) {
+	log.Printf("Error: %w", err)
```

Comment:
При выводе `%v`, а при оборачивании ошибки `%w`
```suggestion
	log.Printf("Error: %v", err)
```